File: javapatterns/src/main/java/org/iut/nantes/strategy/crypto/MD5.java
package org.iut.nantes.strategy.crypto;

import javax.xml.bind.DatatypeConverter;
import java.security.*;
import java.math.*;

public class MD5 implements CrypteStrategy
{
  private MessageDigest m;

  public MD5() throws Exception
  {
    m = MessageDigest.getInstance( "MD5" );
  }

  public String encode(String s)
  {
    // m.update( s.getBytes() );
    // return new String( m.digest() );
    return DatatypeConverter.printHexBinary( m.digest( s.getBytes() ) );
  }
}

File: src/fr/tblf/patterns/strategy/crypto/comportements/MD5.kt
package fr.tblf.patterns.strategy.crypto.comportements

import java.security.MessageDigest

class MD5 : CryptoStrat {
    private val messageDigest: MessageDigest = MessageDigest.getInstance("MD5")

    override fun encrypt(message: String) {
        println(String(messageDigest.digest(message.toByteArray())))
    }
}